tls/s2n_config.h

```diff
@@ -32,6 +32,10 @@ struct s2n_cipher_preferences;
 
 struct s2n_config {
     struct s2n_dh_params *dhparams;
+    /* Needed until we can deprecate s2n_config_add_cert_chain_and_key. This is
+     * used to release memory allocated only in the deprecated API that the application 
+     * does not have a reference to. */
+    struct s2n_cert_chain_and_key *default_cert_and_key_pair;
```

Comment:
nit: if we only plan to use this pointer as a signal that `cert_and_key_pairs` is allocated internally by s2n_config, should we replace this with a boolean-style variable? `is_allocated`?

Reply:
We also use it to keep track of the memory allocated in `s2n_config_add_cert_chain_and_key` so it can later be released in `s2n_config_free`. We can't get rid of it, but are you advocating to add a boolean variable in addition to?